docs/structured-streaming-programming-guide.md

```diff
@@ -2812,6 +2812,19 @@ See [Input Sources](#input-sources) and [Output Sinks](#output-sinks) sections f
 
 # Additional Information
 
+**Notes**
```

Comment:
I was thinking adding this information somewhere API or configuration only. For instance, notes like https://github.com/apache/spark/pull/19617. 

> lots of wondering around SO and user mailing list,

I don't object to note these stuff but usually the site has only key points for some features or configurations. 

If there are more instance to describe specifically for structured streaming (where the same SQL configurations could lead to some confusions), I am fine with adding this. If not or less sure for now, I would add them into API's doc or configuration's doc.